src/v/datalake/record_multiplexer.cc

```diff
@@ -71,11 +79,13 @@ record_multiplexer::operator()(model::record_batch batch) {
         }
 
         auto record_data_res = co_await record_translator::translate_data(
+          _ntp.tp.partition,
           offset,
           std::move(key),
```

Comment:
Similar note about distinguishing between `null` and empty string key. This can be a followup. Key_len should be negative for null. See: https://github.com/twmb/franz-go/blob/81ceb1a064be5de7659e11710fb9d1ee4d27f8a7/pkg/kbin/primitives.go#L386-L393

Reply:
Thanks, will update in a separate PR